tests/utils/test_cycle_consistency.py

```diff
@@ -0,0 +1,299 @@
+        v_corr_idxs=np.array([]),  # dummy array
+        num_inliers_est_model=10,  # dummy value
+        R_error_deg=5,
+        U_error_deg=0,
+    )
+
+    two_view_reports_dict[(0, 2)] = TwoViewEstimationReport(
+        v_corr_idxs=np.array([]),  # dummy array
+        num_inliers_est_model=10,  # dummy value
+        R_error_deg=0,
+        U_error_deg=0,
+    )
+
+    two_view_reports_dict[(2, 4)] = TwoViewEstimationReport(
+        v_corr_idxs=np.array([]),  # dummy array
+        num_inliers_est_model=10,  # dummy value
+        R_error_deg=0,
+        U_error_deg=0,
+    )
+
+    cycle_error, max_rot_error, max_trans_error = cycle_utils.compute_cycle_error(
+        i2Ri1_dict, cycle_nodes, two_view_reports_dict
+    )
+
+    assert np.isclose(cycle_error, 5)
+    assert np.isclose(max_rot_error, 5)
+    assert max_trans_error == 0
+
+
+def test_compute_cycle_error_unknown_GT() -> None:
+    """Ensure cycle error is computed correctly within a triplet, when ground truth is known.
+
+    Imagine 3 poses, all centered at the origin, at different orientations.
+
+    Ground truth poses:
+       Let i0 face along +x axis (0 degrees in yaw)
+       Let i2 have a 30 degree rotation from the +x axis.
+       Let i4 have a 90 degree rotation from the +x axis.
+
+    However, suppose one edge measurement is corrupted (from i0 -> i4) by 5 degrees.
+    """
+    i2Ri0 = Rot3(Rotation.from_euler("y", 30, degrees=True).as_matrix())
+    i4Ri2 = Rot3(Rotation.from_euler("y", 60, degrees=True).as_matrix())
+    i4Ri0 = Rot3(Rotation.from_euler("y", 95, degrees=True).as_matrix())
+
+    cycle_nodes = [0, 2, 4]
+    i2Ri1_dict = {
+        (0, 2): i2Ri0,  # edge i
+        (2, 4): i4Ri2,  # edge j
+        (0, 4): i4Ri0,  # edge k
+    }
+
+    two_view_reports_dict = {}
+    # rest of attributes will default to None
+    two_view_reports_dict[(0, 4)] = TwoViewEstimationReport(
+        v_corr_idxs=np.array([]),  # dummy array
+        num_inliers_est_model=10,  # dummy value
+    )
+
+    two_view_reports_dict[(0, 2)] = TwoViewEstimationReport(
+        v_corr_idxs=np.array([]),  # dummy array
+        num_inliers_est_model=10,  # dummy value
+    )
+
+    two_view_reports_dict[(2, 4)] = TwoViewEstimationReport(
+        v_corr_idxs=np.array([]),  # dummy array
+        num_inliers_est_model=10,  # dummy value
+    )
+
+    cycle_error, max_rot_error, max_trans_error = cycle_utils.compute_cycle_error(
+        i2Ri1_dict, cycle_nodes, two_view_reports_dict
+    )
+
+    assert np.isclose(cycle_error, 5)
+    assert max_rot_error is None
+    assert max_trans_error is None
+
+
+def test_filter_to_cycle_consistent_edges() -> None:
+    """Ensure correct edges are kept in a 2-triplet scenario.
+
+    Scenario Ground Truth: consider 5 camera poses in a line, connected as follows, all with identity rotations:
+
+      _________    ________
+     /         \\ /        \
+    i4 -- i3 -- i2 -- i1 -- i0
```

Comment:
the fact that they are on a line is irrelevant. A better visualization might be two triangles. 
```
i4                 i1
           i2
i3                 i0
```

Reply:
How about I add both figures in the docstring? One that shows the spatial layout, and one that shows the topology?

Reply:
But the spatial layout is irrelevant no? We only care about rotations and that is not evident from it.

Reply:
I think the spatial layout is helpful too. I would probably prefer to keep both

Reply:
Sure, its just a nit. I found the topology useful as this is a rotation-only problem.